Secrets 🔐 PR 7 - Enable Postgres driver to use secrets for configuring SSL parameters

jeff303 opened the pull request:
Adding secret related properties for SSL options to Postgres driver

Move `conn-props->secret-props-by-name` to secret.clj since it's needed directly there, too

Add `us-east-2-bundle.pem` to `test-resources/certificates` for testing Postgres with SSL connectivity to our RDS instance (and a README.md explaining how it differs from the existing `ssl` directory)

Updating `value->file!` to have better logic for building the error message when the existing file is not found (for better UX from the Database admin page)

Updating frontend to support the `visible-if` value being an array, in which case any value may match the current details value, in order for that field to be visible.  Also making the frontend check all entries in the `visible-if` map (not only the first one), and only show the field if all entries in the map are satisfied.

Making the backend traverse all transitive "dependencies" of each `visible-if` entry, to roll up all the dependencies into the leaf node, so that each property is only shown if its visible-if is satisfied, and all its indirectly linked visible-if entries are also satisfied.

Adding secret related properties for SSL options to Postgres driver

Updating CircleCI config.yml to refer to the RDS instance when running Postgres SSL test

jeff303 committed:
Support updating secret values as part of encryption key rotation

Add test assertions for secret values in the `rotate-encryption-key!-test`

jeff303 committed:
Adding test for Oracle connection properties

Adding new `select-keys-sequentially` helper function for dealing with assertions on transformed conn props

jeff303 committed:
Enable Postgres driver to use secrets for configuring SSL parameters

Adding secret related properties for SSL options to Postgres driver

Move `conn-props->secret-props-by-name` to secret.clj since it's needed directly there, too

Add `us-east-2-bundle.pem` to `test-resources/certificates` for testing Postgres with SSL connectivity to our RDS instance (and a README.md explaining how it differs from the existing `ssl` directory)

Updating `value->file!` to have better logic for building the error message when the existing file is not found (for better UX from the Database admin page)

Updating frontend to support the `visible-if` value being an array, in which case any value may match the current details value, in order for that field to be visible

Adding secret related properties for SSL options to Postgres driver

Updating CircleCI config.yml to refer to the RDS instance when running Postgres SSL test

jeff303 committed:
Update translate-conn-props-server->client-test to account for the fact that display name is moved to the -options field now

jeff303 committed:
Fix eastwood issues

jeff303 committed:
Fix conditional

jeff303 committed:
Implement server side expansion of transitive visible-if dependency chain

Update shouldShowEngineProvidedField on client to consider multiple key/value pairs in the visible-if map, and only show the field if ALL are true

Adding new test to confirm transitive visible-if resolution, and cycle detection

jeff303 committed:
Add Cypress test for SSL field visibility

jeff303 committed:
Fix merge conflict destruction

jeff303 committed:
Remove only from Cypress test - oops

daltojohnso commented on frontend/src/metabase/entities/databases/forms.js:
```diff
@@ -216,11 +216,24 @@ function shouldShowEngineProvidedField(field, details) {
+
+    // reduce over all entries in the visible-if map, and only show this field if all key/values are satisfied
+    // (i.e. boolean AND)
+    return Object.entries(detailAndValueRequiredToShowField).reduce(
```
Can you use array's `every` method here in place of `reduce`? Then you won't need to handle the `previousValue`.

jeff303 replied:
Switched

jeff303 committed:
PR feedback

switch reduce to every

daltojohnso reviewed (approved):
FE LGTM

jeff303 committed:
Merge branch 'master' into secrets-7-support-postgres-ssl-modes

jeff303 committed:
Fix merge problem

Add in new arg to connection-props-server->client calls (nil is fine for the tests)
